include/deal.II/base/aligned_vector.h

```diff
@@ -1412,6 +1464,22 @@ AlignedVector<T>::reserve(const size_type new_allocated_size)
 
 
 
+template <class T>
+inline void
+AlignedVector<T>::shrink_to_fit()
+{
+#  ifdef DEBUG
+  Assert(replicated_across_communicator == false,
+         ExcAlignedVectorChangeAfterReplication());
+#  endif
```

Comment:
I guess we don't need to guard this. Assert is only performed in debug mode.

Reply:
In `FEEvaluationData` the `Assert`s are also guarded by `#ifdef DEBUG`. I can remove it if it is not necessary.

Reply:
OK. Let's keep it this way to be consistent.